test/isochrone.cc

```diff
@@ -192,7 +192,7 @@ TEST(Isochrones, Basic) {
4858,52.100474],[5.153663,52.102937],[5.154321,52.103593],[5.156321,52.103359],[5.156892,52.104365],[5.161842,52.104937],[5.157321,52.105503],[5.156749,52.104508],[5.154045,52.104661],[5.152665,52.108281],[5.148744,52.11336],[5.148559,52.117175],[5.146687,52.117303],[5.147321,52.119762],[5.149593,52.117937],[5.149478,52.119094],[5.146387,52.121003],[5.146353,52.122904],[5.147322,52.122935],[5.147321,52.124034],[5.147316,52.122941],[5.146302,52.122937],[5.145993,52.120264],[5.142638,52.119937],[5.146081,52.119697],[5.146024,52.116937],[5.147915,52.113342],[5.145043,52.113214],[5.142321,52.111362],[5.144917,52.109937],[5.142059,52.108198],[5.141321,52.106359],[5.140321,52.109125],[5.134,52.107257],[5.131323,52.107935],[5.129321,52.104612],[5.127463,52.105937],[5.125486,52.105102],[5.121321,52.106482],[5.114321,52.106649],[5.112106,52.105937],[5.111321,52.10454],[5.110532,52.108148],[5.109321,52.108433],[5.108321,52.107485],[5.107321,52.10955],[5.106321,52.109011],[5.103193,52.110064],[5.101321,52.106806],[5.099492,52.108108],[5.099221,52.109837],[5.094522,52.108736],[5.094321,52.107871],[5.092536,52.110152],[5.090343,52.110959],[5.079113,52.113144],[5.078884,52.111937],[5.080159,52.110937],[5.079284,52.109937],[5.08185,52.108466],[5.083025,52.106641],[5.087321,52.107422],[5.089635,52.105623],[5.085321,52.103302],[5.084321,52.104615],[5.082878,52.104494],[5.0826,52.106216],[5.079349,52.105965],[5.078321,52.107218],[5.077321,52.107047],[5.077321,52.104012],[5.078321,52.104623],[5.078929,52.103937],[5.078321,52.103167],[5.077321,52.103618],[5.077321,52.102763],[5.078312,52.101928],[5.080321,52.102151],[5.081321,52.100616],[5.082321,52.101395],[5.082798,52.099937],[5.079321,52.099442],[5.078321,52.100513],[5.077321,52.098449],[5.076321,52.099213],[5.071321,52.098888],[5.071321,52.095681],[5.070164,52.09878],[5.061846,52.106462],[5.060234,52.10685],[5.05533,52.110937],[5.059321,52.112546],[5.062321,52.109518],[5.063768,52.109489],[5.063819,52.111937],[5.061808,52.112424],[5.057766,52.116937],[5.058695,52.118311],[5.056869,52.118485],[5.056386,52.119871],[5.058321,52.119782],[5.05897,52.118586],[5.06122,52.117836],[5.064004,52.117937],[5.062426,52.118042],[5.062321,52.1191],[5.060321,52.12016],[5.056281,52.119976],[5.054321,52.11823],[5.05036,52.118976],[5.050022,52.117235],[5.0475,52.117116],[5.047321,52.117766],[5.047321,52.11617],[5.049321,52.114042],[5.049475,52.114782],[5.052321,52.114542],[5.053321,52.115561],[5.055664,52.11528],[5.055612,52.113645],[5.053321,52.11155],[5.049195,52.112811],[5.049212,52.113828],[5.046321,52.11377],[5.043321,52.115486],[5.041321,52.115474],[5.03587,52.118486],[5.038321,52.121582],[5.038623,52.119937],[5.039321,52.119889],[5.040321,52.120767],[5.041756,52.119937],[5.04205,52.120937],[5.038526,52.122142],[5.038414,52.122937],[5.038489,52.123768],[5.039656,52.123937],[5.041388,52.125869],[5.041476,52.126781],[5.042321,52.126912],[5.042321,52.127328]],"type":"LineString"},"type":"Feature"}],"type":"FeatureCollection"})";
```

Comment:
the difference in isochrones is less clear than I hoped for, i.e. it doesn't seem like the new isochrone is _always_ tighter where it's neighboring NODATA values. the only change that I could visually make out is

<img width="253" height="323" alt="Image" src="https://github.com/user-attachments/assets/8b69c1e1-9792-4b1e-875b-bcb52f4cd23e" />

blue is branch, red is master. IIRC that corner was very often the source of changes..

there's only 1 place where we look at NODATA/`max_value_` in the contouring algo:

https://github.com/valhalla/valhalla/blob/850b2a47a2c68bdf0623065af7d688455f9be603/valhalla/midgard/gridded_data.h#L263-L273

I can't really judge that though.. in the end I don't think we'll create worse shapes than before, just somewhat slightly different.